File: lang/java/src/fun/vac/util/SequenceBuilder.java
package fun.vac.util;

import fun.vac.util.RandomFactory;
import fun.vac.util.Sequences;

public class SequenceBuilder {

    private SequenceBuilder() {}

    public static void packIncreasing(Integer[] arr) {
        RandomFactory.seed();
        arr[0] = RandomFactory.integerN(1, 4);
        for (int i = 1, length = arr.length; i < length; i++) {
            arr[i] = arr[i - 1] + RandomFactory.integerN(1, 4);
        }
    }

    public static void packRandom(Integer[] arr) {
        RandomFactory.seed();
        for (int i = 0, length = arr.length; i < length; i++) {
            arr[i] = RandomFactory.generateInteger();
        }
    }

    public static void packDecreasing(Integer[] arr) {
        packIncreasing(arr);
        Sequences.reverse(arr);
    }
}

File: lang/go/src/fun/vac/util/SequenceBuilder/SequenceBuilder.go
package SequenceBuilder

import (
    "fun/vac/util/RandomFactory"
    "fun/vac/util/Sequences"
)

func PackIncreasing(arr []int) {
    RandomFactory.Seed()
    arr[0] = RandomFactory.IntegerN(1, 3)
    for i, length := 1, len(arr); i < length; i++ {
        arr[i] = arr[i - 1] + RandomFactory.IntegerN(1, 3)
    }
}

func PackRandom(arr []int) {
    RandomFactory.Seed()
    for i := range arr {
        arr[i] = RandomFactory.GenerateInteger()
    }
}

func PackDecreasing(arr []int) {
    PackIncreasing(arr)
    Sequences.Reverse(arr)
}
